docs/sections/user_guide/yaml/execution.rst

```diff
@@ -0,0 +1,76 @@
+.. _execution_yaml:
+
+The ``execution:`` Block:
+=========================
+
+Component drivers require an ``execution:`` block to provide details about how to configure a component's run-time environment, and how to run it.
+
+Example block:
+
+.. code-block:: yaml
+
+   execution:
+     batchargs:
+       export: NONE
+       nodes: 1
+       stdout: /path/to/runscript.out
+       walltime: 00:02:00
+     envcmds:
+       - module use /path/to/modulefiles
+       - module load some_module
+     executable: /path/to/sfc_climo_gen
+     mpiargs:
+       - "--export=ALL"
+       - "--ntasks $SLURM_CPUS_ON_NODE"
+     mpicmd: srun
+
+
+batchargs:
+""""""""""
+
+These entries map to job-scheduler directives sent to e.g. Slurm when a batch job is submitted via the ``--batch`` CLI switch or the ``batch=True`` API argument. The only **required** entry is ``walltime``.
+
+Shorthand names are provided for certain directives for each scheduler, and can be specified as-so along with appropriate values. Recognized names for each scheduler are:
+
+* LSF: ``jobname``, ``memory``, ``nodes``, ``queue``, ``shell``, ``stdout``, ``tasks_per_node``, ``threads``, ``walltime``
+* PBS: ``debug``, ``jobname``, ``memory``, ``nodes``, ``queue``, ``shell``, ``stdout``, ``tasks_per_node``, ``threads``, ``walltime``
+* Slurm: ``cores``, ``exclusive``, ``export``, ``jobname``, ``memory``, ``nodes``, ``partition``, ``queue``, ``rundir``, ``stderr``, ``stdout``, ``tasks_per_node``, ``threads``, ``walltime``
+
+Other, arbitrary directive key-value pairs can be provided exactly as they should appear in the batch runscript. For example
+
+.. code-block:: yaml
+
+   --nice: 100
+
+could be specified to have the Slurm directive
+
+.. code-block: text
+
+   #SBATCH --nice=100
+
+included in the batch runscript.
+
+envcmds:
+""""""""
+
+An **array** (YAML sequence) of commands to execute in the run-time shell before the component executable is run.
+
+executable:
+"""""""""""
+
+The name of or path to the component's executable.
+
+mpiargs:
+""""""""
+
+An **array** (YAML sequence) of string arguments that should follow the MPI launch program (``mpiexec``, ``srun``, et al.) on the command line.
+
+mpicmd:
+"""""""
+
+The MPI launch program (``mpiexec``, ``srun``, et al.)
+
+threads:
```

Comment:
The `threads:` entry under `batchargs:` maps to a specific directive for the job scheduler. For example, setting `threads: 8` under `batchargs:` will result in an `#SBATCH --cpus-per-task=8` directive being set for Slurm, if that's the scheduler in use. This is an instruction to the batch system; it influences the resource allocation created for the batch job. I'm not sure if it has any effect on the actual eventual execution of the component program (maybe if it's executed via `srun`, but probably not otherwise).

The `threads: 8` entry in the `execution:` block maps to an `OMP_NUM_THREADS=8` environment variable being set in the `runsncript`, which directly affects how OpenMP threading of the executable program behaves. FV3 supports OpenMP threading, so it would be sensible to set `OMP_NUM_THREADS` for that program; `sfc_climo_gen` does not AFAIK support OpenMP threads, so it wouldn't make sense (but would do no harm) to set that variable for this component.

We could potentially consider renaming one of these variables to be cleared, but would probably need to understand the interplay, if any, between the two thread-specification mechanisms first.

Reply:
It would be helpful to include an example of each of these uses in the `Example block:` at the top of the page.

Reply:
Added.

Reply:
I'm sorry to drive this one into the ground, but does the final solution also need a `threads: 8` at the `execution:` level? The docs here, and your GH comment point toward yes. How do we DRY that in this case?

Reply:
Thank you for not letting this slip through the cracks. I'm adding the non-`batchargs` version of `threads` to the YAML and updating the description of it in the text below to be more specific about what it does.

As for DRY, I think that -- if it's even possible ("it" being "eliminating at least one of these YAML entries") -- would also involve code changes outside the scope of this PR, and would necessitate knowing if/how `#SBATCH --cpus-per-task` interacts with `OMP_NUM_TASKS`, in which contexts, for which schedulers. I mean, it may be that we need to keep both of these `threads` options if they effect different behaviors that users may need control over.